catalog/resource_grant_test.go

```diff
@@ -0,0 +1,732 @@
+package catalog
+
+import (
+	"testing"
+
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/qa"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGrantPermissionsCornerCases(t *testing.T) {
+	qa.ResourceCornerCases(t, ResourceGrant(), qa.CornerCaseID("schema/sandbox/me"))
+}
+
+func TestResourceGrantCreate(t *testing.T) {
```

Comment:
This is really nice that this resource works even if there are existing permissions for a user on the securable.